intermediate-python-testing.md

```diff
@@ -300,11 +302,72 @@ def test_do_a_thing(mocker):
 
 Tantalizing, right? 
 
-You can also use `mock` to raise exceptions to test error handling, return canned website responses without hitting a live endpoint (as we do [in Metro](https://github.com/datamade/la-metro-councilmatic/blob/master/tests/test_events.py#L117)), or simply turn off state-altering parts of your code that aren't relevant to the test at hand. Finally, `mock` keeps track of whether and how mocked methods are called, so you can test how your code is used (called _n_ times, or with this or that argument), without necessarily having to run it.
+### Getting started
 
-For more on how to use `mock`, see [the quickstart](https://docs.python.org/3/library/unittest.mock.html#quick-guide) in the `mock` documention, as well as [this excellent tutorial](https://www.toptal.com/python/an-introduction-to-mocking-in-python). Meanwhile, here are a few of our own hard-won lessons from early `mock` use.
+#### `patch()`
+
+`patch()` is the main method that allows you to replace a dependency with a mock. There are two main ways to use this method -- as a decorator or as a context manager.
+
+As a decorator, `@patch()` can be used on a testing class or a single test function. The mock that it creates will live for the duration of the class or function. Once the test (or all the tests in the class, as the case may be) is complete, the mock is cleaned up and any other tests will use the real dependency instead of the mock. For example, as a class decorator:
+
+```python
+@patch('path.to.patched_dependency')
+class SomeClass:
+    def first_test(self, patched_dependency):
+        # do the things in the first test using the mock
+        pass
+
+    def second_test(self, patched_dependency):
+        # do the things in the second test using the mock
+        pass
+```
+
+Or as a method decorator:
+```python
+
+class SomeClass:
+    @patch('path.to.patched_dependency')
+    def first_test(self, patched_dependency):
+        # do the things in the first test using the mock
+        pass
+```
+
+As a context manager, `patch()` allows your mock to be scoped to just part of a function, like this:
+```python
+def first_test(self):
+    patched_dependency.return_value = "this is what I expect"
```

Comment:
Seems like this example may be missing a line that defines `patched_dependency`?